web_drop_target/static/src/js/web_drop_target.js

```diff
@@ -27,44 +28,67 @@ odoo.define('web_drop_target', function(require) {
         },
 
         _on_drop: function(e) {
-            var drop_item = this._get_drop_item(e);
-            if(!drop_item) {
-                return;
-            }
-            jQuery(e.delegateTarget).removeClass(this._drag_over_class);
-            var reader = new FileReader();
-            reader.onloadend = this.proxy(
-                _.partial(this._handle_file_drop, drop_item.getAsFile())
-            );
-            reader.readAsArrayBuffer(drop_item.getAsFile());
+            var self = this;
+            var drop_items = this._get_drop_item(e);
+            _.each(drop_items, function(drop_item) {
+                var reader = new FileReader();
+                reader.onloadend = self.proxy(
+                    _.partial(self._handle_file_drop, drop_item.getAsFile())
```

Comment:
this is the simple way to implement this, but is it also the best way? That's a serious question, handling files separately makes sense I think, but let's think if there are cases where it makes a difference on the client side if we handle one file or 20? Can't think of one currently, but I wanted to raise the point anyways

Reply:
Yes, you're completely right. I had tested it with 3/4 pdf and worked perfectly but TBH, now I don't know a better way to do it 😅

Reply:
let's leave this as future extension then